streams-contrib/streams-provider-moreover/src/main/java/org/apache/streams/data/moreover/MoreoverProvider.java

```diff
@@ -93,17 +109,19 @@ public StreamsResultSet readRange(DateTime start, DateTime end) {
 
     @Override
     public boolean isRunning() {
-        return !executor.isShutdown() && !executor.isTerminated();
+        return !executor.isTerminated() || !this.providerQueue.isEmpty();
```

Comment:
Is this to catch the case where the ExecutorService.shutdown() method has been called but the last few tasks submitted to the service are still running?